### PR TITLE
Enhance: Importing quantized models after bias correction

### PR DESCRIPTION
**Pull request by costigt-dev**
## Issue
Implements https://github.com/Xilinx/brevitas/issues/827

## Details

- Adds new context manager **_load_quant_model_** that silently adds (and removes if unused) biases to help load any state dictionary that may have had biases unexpectedly added via use of the **_bias_correction_mode_** context manager prior to saving
- Adds flag to **_bias_correction_mode_** and **__BiasCorrection_** to disable the adding of corrections to layers that don't have a bias, default behaviour is the same as before so without modifying the flag, correction will still be applied
- Adds two tests for the above changes

### CODE REVIEW

File: src/brevitas/graph/calibrate.py [hunk 1]
```diff
@@ -85,11 +89,33 @@ def __exit__(self, type, value, traceback):
...
+                if module.bias is None:
+                    module.register_parameter(
+                        'bias',
+                        nn.Parameter(torch.empty(module.weight.shape[0])).to(module.weight.device))
+                    self.tracked_modules.append(module)
+
+    def __exit__(self, type, value, traceback):
+        for module in self.tracked_modules:
+            # empty tensor has a numel result of 0
+            if torch.numel(module.bias) == 0:
```
Comment (Giuseppe5) [inline src/brevitas/graph/calibrate.py:110]: let's check that str(module.device) == 'meta'

File: src/brevitas/graph/calibrate.py [hunk 2]
```diff
@@ -85,11 +89,33 @@ def __exit__(self, type, value, traceback):
...
+                module.bias = None
+
+
 class bias_correction_mode:
 
-    def __init__(self, model, enabled=True):
+    def __init__(self, model, enabled=True, only_layers_with_bias=False):
```
Comment (Giuseppe5) [inline src/brevitas/graph/calibrate.py:116]: rename the flag to:
skip_if_no_bias

File: src/brevitas/graph/calibrate.py (hunk 1 above)
Comment (costigt-dev) [reply src/brevitas/graph/calibrate.py:110]: the approach i've used doesn't use the meta device but rather an empty tensor as the meta tensor seemed to add a lot of complication and warnings

Comment (Giuseppe5) [reply src/brevitas/graph/calibrate.py:110]: If I'm not mistaken, an empty tensor has "meta" as device by default. Or am I missing something?

Comment (costigt-dev) [reply src/brevitas/graph/calibrate.py:110]: My testing has the default as 'cpu'
`torch.empty(10).device` 
gives 'cpu'

Comment (Giuseppe5) [reply src/brevitas/graph/calibrate.py:110]: Out of curiosity, what warnings do you get with meta tensors?

Comment (costigt-dev) [reply src/brevitas/graph/calibrate.py:110]: I didn't go too deep into a meta tensor implementation but warnings like:
> UserWarning: for net.0.bias: copying from a non-meta parameter in the checkpoint to a meta parameter in the current model, which is a no-op.

File: src/brevitas/graph/calibrate.py [hunk 3]
```diff
@@ -24,7 +24,11 @@
 from .base import Transform
 
 __all__ = [
-    'ClipFloatWeights', 'DisableEnableQuantization', 'bias_correction_mode', 'calibration_mode']
+    'ClipFloatWeights',
+    'DisableEnableQuantization',
+    'bias_correction_mode',
+    'calibration_mode',
+    'allow_unexpected_bias_keys']
```
Comment (Giuseppe5) [inline src/brevitas/graph/calibrate.py:31]: Let's rename `allow_unexpected_bias_keys` to `load_quant_model`.
In the future we can use this as a more general entrypoint to load checkpoint with special requirements.

For example, at the moment we have a global flag to ignore missing keys when loading a floating point dict in a quantized model, and it's up to the user to enable that flag. 

This is a matter for another PR anyway.

File: src/brevitas/graph/calibrate.py [hunk 4]
```diff
@@ -85,11 +89,33 @@ def __exit__(self, type, value, traceback):
             self.model, is_training=self.previous_training_state, quantization_enabled=True)
 
 
+class load_quant_model:
+
+    def __init__(self, model):
+        self.model = model
+
+    def __enter__(self):
+        self.tracked_modules = []
```
Comment (Giuseppe5) [inline src/brevitas/graph/calibrate.py:98]: Let's define this in the init

**Pull request merged**